docs/shared/recipes/add-stack/add-svelte.md

```diff
@@ -0,0 +1,288 @@
+{% /callout %}
+
+Update your `tsconfig.lib.json` to be `tsconfig.app.json` and add the following content:
+
+```json
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "moduleResolution": "node",
+    "target": "esnext",
+    "ignoreDeprecations": "5.0",
+    "isolatedModules": true,
+    "sourceMap": true,
+    "types": ["svelte", "node", "vite/client"],
+    "strict": false,
+    "esModuleInterop": true,
+    "skipLibCheck": true,
+    "forceConsistentCasingInFileNames": true,
+    "checkJs": true
+  },
+  "include": [
+    "src/**/*.d.ts",
+    "src/**/*.ts",
+    "src/**/*.js",
+    "src/**/*.svelte",
+    "vite.config.mts"
+  ],
+  "exclude": ["jest.config.ts", "src/**/*.spec.ts", "src/**/*.test.ts"]
+}
+```
+
+Navigate to `project.json` and update it with the following content:
+
+```json
+    "build": {
+      "executor": "@nx/vite:build",
+      "outputs": ["{options.outputPath}"],
+      "defaultConfiguration": "production",
+      "options": {
+        "outputPath": "dist/acme"
+      },
+      "configurations": {
+        "development": {
+          "mode": "development"
+        },
+        "production": {
+          "mode": "production"
+        }
+      }
+    },
+    "serve": {
+      "executor": "@nx/vite:dev-server",
+      "defaultConfiguration": "development",
+      "options": {
+        "buildTarget": "acme:build"
+      },
+      "configurations": {
+        "development": {
+          "buildTarget": "acme:build:development",
+          "hmr": true
+        },
+        "production": {
+          "buildTarget": "acme:build:production",
+          "hmr": false
+        }
+      }
+    },
+```
+
+This allows us to use `nx build` and `nx serve` to build and serve or Svelte application.
+
+Test it out
+
+**Build the application**
+
+```shell
+nx build acme
+```
+
+Your build artifacts should be in `dist/acme`
+
+**Serve the application**
+
+```shell
+nx serve acme
+```
+
+Navigate to `http://localhost:4200` and you should see your application.
+
+## Create a library
+
+Instead of having our Counter directly defined in `App.svelte` file, let's create a library that we can import into our application.
+
+```shell
+nx generate @nx/js:library --name=Counter --unitTestRunner=vitest --bundler=vite --importPath=@acme/counter
+```
+
+Create the Counter component at `counter/src/lib/Counter.svelte` and copy the contents of your `src/App.svelte` file into it.
+
+Update your `libs/counter/src/lib/index.ts` to export your Counter component.
+
+```ts
+export { default as Counter } from './Counter.svelte';
+```
+
+{% callout "Note" %}
+The `default` is very import here as it allows us to import the component using `import { Counter } from '@acme/counter'` instead of `import Counter from '@acme/counter'`.
```

Comment:
Why is this important?  Does Svelte need to have components exported as default?  Or is that the convention in Svelte apps?

Reply:
I think it's based on the alias which is set inside `vite.config.ts` without the `default` the import fails to resolve at run-time